src/devices_models/devices/hydro_generation.jl

```diff
@@ -666,6 +692,7 @@ function initial_conditions!(
     return
 end
 
+# TODO: Add test that checks for this function
```

Comment:
@sourabhdalvi this initial condition call isn't covered in the tests and I am fairly certain it has be incorrectly implemented.

Reply:
I think your correct that this isn't covered because we don't use these initial conditions in the hydro code at all. So we have to discuss, if these are really necessary and which formulation would use them.